pkg/component/handler_test.go

```diff
@@ -0,0 +1,1764 @@
+				if err != nil {
+					t.Error("error reading config")
+				}
+				ctx = envcontext.WithEnvConfig(ctx, *envConfig)
+				devfileObj := tt.devfileObjRun()
+				cmdHandler := &runHandler{
+					ctx:                   ctx,
+					fs:                    filesystem.NewFakeFs(),
+					execClient:            tt.execClient(ctrl),
+					platformClient:        tt.platformClient(ctrl),
+					configAutomountClient: tt.configAutomountClient(ctrl),
+					imageBackend:          tt.imageBackend(ctrl),
+					devfile:               devfileObj,
+				}
+				err = libdevfile.ExecuteCommandByNameAndKind(ctx, devfileObj, "", v1alpha2.RunCommandGroupKind, cmdHandler, false)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("Err expected %v, got %v", tt.wantErr, err)
+				}
+			}
+			{
+				ctrl := gomock.NewController(t)
+				ctx := context.Background()
+				ctx = odocontext.WithDevfilePath(ctx, "/devfile.yaml")
+				ctx = odocontext.WithApplication(ctx, appName)
+				ctx = odocontext.WithComponentName(ctx, componentName)
+				envConfig, err := config.GetConfigurationWith(envconfig.MapLookuper(tt.env))
+				if err != nil {
+					t.Error("error reading config")
+				}
+				ctx = envcontext.WithEnvConfig(ctx, *envConfig)
+				devfileObj := tt.DevfileObjPostStart()
+				cmdHandler := &runHandler{
+					ctx:                   ctx,
+					fs:                    filesystem.NewFakeFs(),
+					execClient:            tt.execClient(ctrl),
+					platformClient:        tt.platformClient(ctrl),
+					configAutomountClient: tt.configAutomountClient(ctrl),
+					imageBackend:          tt.imageBackend(ctrl),
+					devfile:               devfileObj,
+				}
+				err = libdevfile.ExecPostStartEvents(ctx, devfileObj, cmdHandler)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("Err expected %v, got %v", tt.wantErr, err)
+				}
+			}
+			{
+				ctrl := gomock.NewController(t)
+				ctx := context.Background()
+				ctx = odocontext.WithDevfilePath(ctx, "/devfile.yaml")
+				ctx = odocontext.WithApplication(ctx, appName)
+				ctx = odocontext.WithComponentName(ctx, componentName)
+				envConfig, err := config.GetConfigurationWith(envconfig.MapLookuper(tt.env))
+				if err != nil {
+					t.Error("error reading config")
+				}
+				ctx = envcontext.WithEnvConfig(ctx, *envConfig)
+				devfileObj := tt.DevfileObjPreStop()
+				cmdHandler := &runHandler{
+					ctx:                   ctx,
+					fs:                    filesystem.NewFakeFs(),
+					execClient:            tt.execClient(ctrl),
+					platformClient:        tt.platformClient(ctrl),
+					configAutomountClient: tt.configAutomountClient(ctrl),
+					imageBackend:          tt.imageBackend(ctrl),
+					devfile:               devfileObj,
+				}
+				err = libdevfile.ExecPreStopEvents(ctx, devfileObj, cmdHandler)
+				if (err != nil) != tt.wantErr {
+					t.Errorf("Err expected %v, got %v", tt.wantErr, err)
+				}
+			}
+		})
+	}
+}
+
+/*
+func TestApplyImage(t *testing.T) {
```

Comment:
Can we remove these commented lines if not needed? I can see they seem to be covered with the big test above..

Reply:
I forgot to cleanup these old tests. Thanks